queue-jdbc/src/main/java/io/kestra/queue/jdbc/client/JdbcQueueClient.java

```diff
@@ -0,0 +1,198 @@
+        try {
+            dslContextWrapper.transaction(configuration -> {
+                DSLContext context = DSL.using(configuration);
+
+                InsertValuesStepN<Record> insert = context
+                    .insertInto(jdbcRepository.getTable())
+                    .columns(COLUMNS);
+
+                for (Map.Entry<String, String> entry : values.entrySet()) {
+                    insert = insert.values(
+                        queueNameToType(queue),
+                        routingKey,
+                        entry.getKey(),
+                        JSONB.valueOf(entry.getValue()),
+                        Instant.now()
+                    );
+                }
+
+                insert.execute();
+            });
+        } catch (DataException e) { // The exception is from the data itself, not the database/network/driver so instead of fail fast, we throw a recoverable QueueException
+            // Postgres refuses to store JSONB with the '\0000' codepoint as it has no textual representation.
+            // We try to detect that and fail with a specific exception so the Worker can recover from it.
+            if (e.getMessage() != null && e.getMessage().contains("ERROR: unsupported Unicode escape sequence")) {
+                throw new UnsupportedMessageException(e.getMessage(), e);
+            }
+            throw new QueueException("Unable to emit a message to the queue", e);
+        }
+    }
+
+    public Integer subscribeDispatch(String queue, @Nullable String routingKey, MessageConsumer<String, Exception> consumer) {
+        return dslContextWrapper.transactionResult(conf -> {
+            DSLContext context = DSL.using(conf);
+
+            SelectConditionStep<Record> select = context.select(DSL.asterisk())
+                .from(this.jdbcRepository.getTable())
+                .where(io.kestra.jdbc.repository.AbstractJdbcRepository.field("type").eq(queueNameToType(queue)));
+
+            if (routingKey != null) {
+                select = select.and(io.kestra.jdbc.repository.AbstractJdbcRepository.field("routing_key").eq(routingKey));
+            }
+
+            List<JdbcQueueItem> queueItems = select
+                .orderBy(io.kestra.jdbc.repository.AbstractJdbcRepository.field("offset").asc())
+                .limit(configuration.pollSize())
+                .forUpdate()
+                .skipLocked()
+                .fetchInto(JdbcQueueItem.class);
+
+            if (!queueItems.isEmpty()) {
+                List<Long> processedItems = queueItems
+                    .stream()
+                    .map(queueItem -> {
+                        Exception exception = consumer.apply(queueItem.getValue());
+                        return exception == null ? queueItem.getOffset() : null;
+                    })
+                    .filter(Objects::nonNull)
+                    .toList();
+
+                if (!processedItems.isEmpty()) {
```

Comment:
Here, if an exception occur we would not consume the item so it would becomes a poison pill.
Is it really what we want? In the previous JDBC impl it was not the case